packages/data-stores/src/reader/mutations/use-post-notify-me-of-new-comments-mutation.ts

```diff
@@ -0,0 +1,89 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { callApi } from '../helpers';
+import { useCacheKey, useIsLoggedIn } from '../hooks';
+import { postSubscriptionsQueryKeyPrefix } from '../queries/use-post-subscriptions-query';
+import type { PostSubscriptionsResult } from '../types';
+
+type PostSubscriptionNotifyMeOfNewCommentsParams = {
+	subscriptionId: string;
+	sendComments: boolean;
+};
+
+type PostSubscriptionNotifyMeOfNewCommentsResponse = {
+	updated: boolean;
+};
+
+const usePostNotifyMeOfNewCommentsMutation = () => {
+	const { isLoggedIn } = useIsLoggedIn();
+	const queryClient = useQueryClient();
+	const postSubscriptionsCacheKey = useCacheKey( postSubscriptionsQueryKeyPrefix );
+
+	return useMutation( {
+		mutationFn: async ( params: PostSubscriptionNotifyMeOfNewCommentsParams ) => {
+			if ( ! params.subscriptionId || typeof params.sendComments !== 'boolean' ) {
+				throw new Error(
+					'Something went wrong while changing the "Notify me of new comments" setting.'
+				);
+			}
+
+			const response = await callApi< PostSubscriptionNotifyMeOfNewCommentsResponse >( {
+				apiNamespace: 'wpcom/v2',
+				path: `/post-comment-subscriptions/${ params.subscriptionId }/update`,
+				method: 'POST',
+				body: {
+					send_comments: params.sendComments,
+				},
+				isLoggedIn,
+				apiVersion: '1.1',
```

Comment:
Mm... I think the version here should be 2.

Reply:
Good catch, I'll update it. Thanks!